### PR TITLE
Export list of exercices in a file "info.yml"

### PR DESCRIPTION
**Pull request by ohmushi**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,25 @@
-cmake_minimum_required(VERSION 3.27)
+cmake_minimum_required(VERSION 3.26)
```
Comment (Trevodorax) [inline CMakeLists.txt:1]: why tho ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,11 +1,25 @@
...
 set(CMAKE_C_STANDARD 11)
 
+find_package(yaml CONFIG REQUIRED)
+
+include_directories(src)
+
 add_executable(clings main.c
         src/execution/execution.c
+        src/execution/execution.h
+        src/types/types.h
```
Comment (Trevodorax) [inline CMakeLists.txt:13]: I thought you wanted to remove the .h from here ?

File: src/katas/katas.h [hunk 3]
```diff
@@ -1,17 +1,34 @@
 #ifndef CLINGS_KATAS_H
 #define CLINGS_KATAS_H
 
+#include "types/types.h"
+#include "stdbool.h"
 
-// TODO: extract this to yaml
-static const char * kata_folder_path = "../katas/";
+typedef struct Kata Kata;
+struct Kata {
```
Comment (Trevodorax) [inline src/katas/katas.h:8]: We need a convention for types

either we do
types_like_this_t
or 
TypesLikeThis

I'm okay with the one you did, but then we should use camelCase for variable names shouldn't we ?

File: src/katas/katas.h [hunk 4]
```diff
@@ -1,17 +1,34 @@
...
+    sized_string_t name;
+    sized_string_t path;
+};
+
+typedef struct KataList KataList;
```
Comment (Trevodorax) [inline src/katas/katas.h:13]: some here for the convention (and on all other types)

File: src/yaml/yml.h [hunk 5]
```diff
@@ -0,0 +1,12 @@
+
+
```
Comment (Trevodorax) [inline src/yaml/yml.h:2]: Empty lines

**Review by Trevodorax — CHANGES_REQUESTED** [review]
Super joli 😍

File: CMakeLists.txt (hunk 1 above)
Comment (ohmushi) [reply CMakeLists.txt:1]: mon clion a un cmake bundled à 3.26 et j'arrive pas à l'upgrade... T_T

File: CMakeLists.txt (hunk 2 above)
Comment (ohmushi) [reply CMakeLists.txt:13]: à un oubli

File: src/katas/katas.h (hunk 3 above)
Comment (ohmushi) [reply src/katas/katas.h:8]: Finally I think we should use camelCase

File: info.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+exercises:
+  - name: intro
+    path: katas/intro.c
```
Comment (Trevodorax) [inline info.yml:3]: couldn't we remove the folder name from here ? or maybe have a kata_folder variable somewhere in the yml ?

File: main.c [hunk 7]
```diff
@@ -1,5 +1,5 @@
 #include "src/clings.h"
 
-int main() {
+int main(void) {
```
Comment (Trevodorax) [inline main.c:3]: why not int argc, char ** argv ?

File: src/clings.h [hunk 8]
```diff
@@ -2,6 +2,6 @@
 #ifndef CLINGS_CLINGS_H
 #define CLINGS_CLINGS_H
 
-int clings();
+int clings(void);
```
Comment (Trevodorax) [inline src/clings.h:5]: why add void ?

File: src/katas/katas.h [hunk 9]
```diff
@@ -1,17 +1,31 @@
...
+        kata_list_t kata_list;
+        char *error_message;
+    };
+} kata_list_parsing_result_t;
+
+void free_kata_list(kata_list_t *kata_list);
+
+void push_kata_in_list(kata_t kata, kata_list_t *list);
+
+kata_t no_kata(void);
```
Comment (Trevodorax) [inline src/katas/katas.h:29]: Would be nice to have doxygen on every exposed functions

File: src/types/sized_string.c [hunk 10]
```diff
@@ -17,4 +18,19 @@ void free_sized_string(sized_string_t * string) {
         string->str = NULL;
     }
     string->len = 0;
-}
\ No newline at end of file
+}
+
+sized_string_t copy_str_to_sized_string(char * str, size_t len) {
+    sized_string_t s;
```
Comment (Trevodorax) [inline src/types/sized_string.c:24]: Maybe a clearer variable name

File: src/types/types.h [hunk 11]
```diff
@@ -25,6 +25,10 @@ typedef struct sized_string_t {
  */
 sized_string_t new_sized_string(size_t len);
 
+sized_string_t copy_str_to_sized_string(char * str, size_t len);
+
+sized_string_t empty_sized_string(void);
```
Comment (Trevodorax) [inline src/types/types.h:30]: Some doxygen por favor :D

File: src/yaml/yaml_parser_factory.c [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+yaml_parser_t parser_factory(char * file_path) {
+    FILE *input = fopen(file_path, "rb");
+    if (input == NULL) {
+        return (yaml_parser_t) {
+                .error = YAML_READER_ERROR,
+                .problem = "Failed to open yaml file."
+        };
+    }
+    yaml_parser_t parser;
+    if(yaml_parser_initialize(&parser))
```
Comment (Trevodorax) [inline src/yaml/yaml_parser_factory.c:12]: is it ok for you if we have brackets around all "if" ? I think it's clearer

**Review by Trevodorax — CHANGES_REQUESTED** [review]

File: main.c (hunk 7 above)
Comment (ohmushi) [reply main.c:3]: I have enabled the compiler flags `-Wall -Wextra -Werror -pedantic`

Now compilation is a little more rigid. Here, the `-Werror` flag embeds the `-Wunused-parameter` flag, which causes compilation to fail in the event of unused parameters.

File: src/clings.h (hunk 8 above)
Comment (ohmushi) [reply src/clings.h:5]: The flag `-Werror` also embeds the flag `-Wstrict-prototypes` which come with a sweet compilation message `error: a function declaration without a prototype is deprecated in all versions of C  [-Werror,-Wstrict-prototypes]`.

So for now the prototypes have to have a void parameter for no parameters.

File: src/types/types.h (hunk 11 above)
Comment (ohmushi) [reply src/types/types.h:30]: Si Señor

File: src/types/sized_string.c (hunk 10 above)
Comment (ohmushi) [reply src/types/sized_string.c:24]: I'll go for `string`

File: info.yml (hunk 6 above)
Comment (ohmushi) [reply info.yml:3]: I actually copied this structure from Rustlings, and honestly I think adding this is a lot of work for not so mush value.

File: src/types/sized_string.c [hunk 13]
```diff
@@ -21,10 +21,10 @@ void free_sized_string(sized_string_t * string) {
 }
 
 sized_string_t copy_str_to_sized_string(char * str, size_t len) {
```
Comment (TheRealHoko) [inline src/types/sized_string.c:23]: could be simpler ? by reusing new_sized_string function and strncpy after

**Review by Trevodorax — APPROVED** [review]
Awesome

File: src/types/sized_string.c (hunk 13 above)
Comment (ohmushi) [reply src/types/sized_string.c:23]: Yeah you're right. I will change that

Comment (ohmushi) [reply src/types/sized_string.c:23]: I changed it, can you check ?

**Comment by TheRealHoko** [comment]
Nice to add lenght to param

**Review by TheRealHoko — APPROVED** [review]

**Pull request merged**